### PR TITLE
Add more information (sortdate, orig_uid, orig_pid, _languageUid, and…

### PR DESCRIPTION
**Pull request by mrvoss**
… system categories as tags) to the ke_search index

### CODE REVIEW

File: Classes/Hooks/KeSearchIndexer.php [hunk 1]
```diff
@@ -70,13 +70,35 @@ public function customIndexer(&$indexerConfig, &$indexerObject)
             $abstract = strip_tags($originalObject->getKeSearchAbstract($index));
             $content = strip_tags($originalObject->getKeSearchContent($index));
             $fullContent = $title . "\n" . $abstract . "\n" . $content;
+            $tags = [];
+            foreach ($originalObject->getCategories() as $category) {
```
Comment (lochmueller) [inline Classes/Hooks/KeSearchIndexer.php]: Please take care, that not every Event model use categorization. Please check with is_callable and $category with instanceof Category.

File: Classes/Hooks/KeSearchIndexer.php [hunk 2]
```diff
@@ -70,13 +70,35 @@ public function customIndexer(&$indexerConfig, &$indexerObject)
...
+            $indexerObject->storeInIndex(
+                $indexerConfig['storagepid'],
+                $title,
+                'calendarize',
+                $indexerConfig['targetpid'],
+                $fullContent,
+                $tags,
+                "&tx_calendarize_calendar[index]={$index->getUid()}",
+                $abstract,
+                $index->_getProperty('_languageUid'),
```
Comment (lochmueller) [inline Classes/Hooks/KeSearchIndexer.php]: Please add also a check for translation. Not every Event structure use l10n features...

**Comment by lochmueller** [comment]
Plase take a look for the two comments. The index should be usable for any kind of Event structures...

File: Classes/Hooks/KeSearchIndexer.php (hunk 1 above)
Comment (mrvoss) [inline Classes/Hooks/KeSearchIndexer.php]: Would it make sense to add `getCategories()` to `KeSearchIndexInterface`? Or `getKeSearchTags()` - which one is better? Looks a little cleaner than `is_callable` and` instanceof Category` to me.

Comment (lochmueller) [inline Classes/Hooks/KeSearchIndexer.php]: You are right `getKeSearchTags()` in the Interface is much cleaner (also add the implementation for the Calendarize Event model). So the developer get also the chance to add tags independent of the category relation....

File: Classes/Hooks/KeSearchIndexer.php (hunk 2 above)
Comment (lochmueller) [inline Classes/Hooks/KeSearchIndexer.php]: Ignore this comment. The index always has a "_languageUid". If the related Event structure do not use translations, the Index still has the sys_language_uid=0. So this line is OK :+1:

**Comment by mrvoss** [comment]
I addressed all your comments and added a little more in my second commit yesterday (did you get a notification for that?)

**Comment by lochmueller** [comment]
Thanks for the second patch... :+1:

**Pull request merged**